### PR TITLE
Add caching around the `url_to_postid()` call.

### PR DESCRIPTION
**Pull request by borkweb**
See: https://github.com/GigaOM/gigaom/issues/5020

### CODE REVIEW

**Comment by Camwyn** [comment]
:whale2: :dash:

File: components/class-go-404.php [hunk 1]
```diff
@@ -50,7 +50,12 @@ public function clean_and_validate_url( $url )
 			return NULL;
 		}
 
-		$post_id = url_to_postid( $url );
+		$post_id = wp_cache_get( $url, 'go-404-url-to-postid' );
+		if ( FALSE === $post_id )
+		{
+			$post_id = url_to_postid( $url );
+			wp_cache_set( $url, 'go-404-url-to-postid' );
```
Comment (misterbisson) [inline components/class-go-404.php]: Um, this isn't right, is it?

**Comment by borkweb** [comment]
Tested:
- `http://go.wp.local.gostage.it/2014/08/05/example-content/'+trackingPixel+'`
- `http://go.wp.local.gostage.it/2014/08/05/example-content/%27`
- `http://go.wp.local.gostage.it/2014/08/05/example-content/'`

All three went to: `http://go.wp.local.gostage.it/2014/08/05/example-content/` after multiple reloads.

File: components/class-go-404.php (hunk 1 above)
Comment (borkweb) [inline components/class-go-404.php]: Yeah, I noticed that too after looking at the PR. Whoopsie.

**Comment by misterbisson** [comment]
:pickles:

**Pull request merged**